artifactory.py

```diff
@@ -1471,8 +1472,13 @@ def copy(self, dst, suppress_layouts=False):
         if self.drive.rstrip("/") == dst.drive.rstrip("/"):
             self._accessor.copy(self, dst, suppress_layouts=suppress_layouts)
         else:
-            with self.open() as fobj:
-                dst.deploy(fobj)
+            with TemporaryDirectory() as tmp_dir:
```

Comment:
If we're copying "well deployed" artifact that was deployed with `deploy_file(..., calc_md5=True, ...)` before, then we shouldn't download whole file cause Artifactory already has precalculated checksums and we get it via HTTP API: https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-FileInfo
```python
"checksums":
{
        "md5" : string,
        "sha1" : string,
        "sha256" : string
    },
"originalChecksums":{
        "md5" : string,
        "sha1" : string,
        "sha256" : string
    }
}
```

There are two solutions for checksums:
1. With "originalChecksum" - we should set them from the `src` artifact
2. Without "originalChecksum" - we should leave them as it is (and get the the error from https://stackoverflow.com/questions/40009688/artifactory-upload-with-checksum)

Reply:
I mean in PR's solution we will have different checksum (`src.checksum != dst.checksum`) if something changes our file after we downloaded it into `tmp` but before `dst.deploy_file`. 
Also we'll eat our disk space. If we want to copy 4GB file - we must download and save it here.

I wouldn't suggest do it, let's try to find solution via Artifactory API

Reply:
That makes sense. I'll investigate it.